File: Java/src/com/leetcode/Easy/MiddleOfTheLinkedList.java
package com.leetcode.Easy;

// 	876. Middle of the Linked List
public class MiddleOfTheLinkedList {
    ListNode head;

    static class ListNode {
        int val;
        ListNode next;

        ListNode(int value) {
            val = value;
            next = null;
        }
    }

    public ListNode middleNode(ListNode head) {
        if (head == null)
            return null;

        int mid = 0, size = 0;
        ListNode curr = head;
        while (curr != null) {
            size++;
            curr = curr.next;
        }

        curr = head;
        for (int i = 0; i < (size / 2); i++) {
            curr = curr.next;
        }
        head = curr;
        return head;
    }
}

File: JavaScript/Easy/MiddleOfTheLinkedList.js
/**
 * 876. Middle of the Linked List
 * @param {ListNode} head
 * @return {ListNode}
 */
const middleNode = function (head) {
  let slow = head, fast = head;
  while (fast != null && fast.next != null) {
    slow = slow.next;
    fast = fast.next.next;
  }

  return slow;
};